statsmodels/robust/covariance.py

```diff
@@ -0,0 +1,1125 @@
+    cs : float
+        constant used in call to weight_scale
+    weight_mean : callable
+        function to calculate weights for weighted mean
+    weight_scale : callable
+        function to calculate scale, "rho" function
+    normalize : bool
+        rescale the scale estimate so it is consistent when the data is
+        normally distributed. The computation assumes winsorized (truncated)
+        variance.
+
+    Returns
+    -------
+    mean : nd_array
+        robust mean
+    std : nd_array
+        robust estimate of scale (standard deviation)
+
+    Notes
+    -----
+    Uses definition of Maronna and Zamar 2002, with weighted mean and
+    trimmed variance.
+    The normalization has been added to match R robustbase.
+    R robustbase uses by default ddof=0, with option to set it to 2.
+
+    References
+    ----------
+    .. [1] Maronna, Ricardo A, and Ruben H Zamar. “Robust Estimates of Location
+       and Dispersion for High-Dimensional Datasets.” Technometrics 44, no. 4
+       (November 1, 2002): 307–17. https://doi.org/10.1198/004017002188618509.
+    """
+
+    x = np.asarray(data)
+    nobs = x.shape[0]
+
+    med_x = np.median(x, 0)
+    xdm = x - med_x
+    mad_x = np.median(np.abs(xdm), 0)
+    wm = weight_mean(xdm / mad_x, cm)
+    mean = (wm * x).sum(0) / wm.sum(0)
+    var = (mad_x**2 * weight_scale((x - mean) / mad_x, cs).sum(0) /
+           (nobs - ddof))
+
+    cf = 1
+    if normalize:
+        c = cs * stats.norm.ppf(0.75)
+        cf = 2 * ((1 - c**2) * stats.norm.cdf(c) - c * stats.norm.pdf(c)
+                  + c**2) - 1
+    # return Holder(loc=mean, scale=np.sqrt(var / cf))
+    return mean, np.sqrt(var / cf)
+
+
+def mahalanobis(data, cov=None, cov_inv=None):
+    """Mahalanobis distance squared
+
+    Note: this is without taking the square root
+
+    """
+    x = np.asarray(data)
+    if cov_inv is not None:
+        d = (x * cov_inv.dot(x.T).T).sum(1)
+    elif cov is not None:
+        d = (x * np.linalg.solve(cov, x.T).T).sum(1)
+    else:
+        raise ValueError('either cov or cov_inv needs to be given')
+
+    return d
+
+
+def cov_gk1(x, y, scale_func=mad):
+    """Gnanadesikan and Kettenring covariance between two variables
+
+    """
+    s1 = scale_func((x + y))
+    s2 = scale_func((x - y))
+    return (s1**2 - s2**2) / 4
+
+
+def cov_gk(data, scale_func=mad):
+    """Gnanadesikan and Kettenring covariance estimator
+
+    uses loop with cov_gk1
+
+    """
+    x = np.asarray(data)
+    if x.ndim != 2:
+        raise ValueError('data needs to be two dimensional')
+    nobs, k_vars = x.shape  # noqa: F841
+    cov = np.diag(scale_func(x)**2)
+    for i in range(k_vars):
+        for j in range(i):
+            cij = cov_gk1(x[:, i], x[:, j], scale_func=scale_func)
+            cov[i, j] = cov[j, i] = cij
+    return cov
+
+
+def cov_ogk(data, maxiter=2, scale_func=mad, cov_func=cov_gk,
+            loc_func=lambda x: np.median(x, axis=0), reweight=0.9,
```

Comment:
the article maronna/zamar uses 0.9 in Monte Carlo and recommends that.
I think for consistency with other methods, we should standardize on 0.975 as default